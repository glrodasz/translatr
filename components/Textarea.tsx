import React from "react";

interface TextareaProps {
  placeholder: string;
  value: string;
  onChange: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
  readOnly?: boolean;
}

const Textarea: React.FC<TextareaProps> = ({
  placeholder,
  value,
  onChange,
  readOnly,
}) => {
  return (
    <textarea
      className="w-full mt-4 p-4 bg-gray-100 dark:bg-gray-700 text-gray-900 dark:text-gray-100 rounded resize-none"
      rows={10}
      value={value}
      onChange={onChange}
      placeholder={placeholder}
      readOnly={readOnly}
    ></textarea>
  );
};

export default Textarea;
