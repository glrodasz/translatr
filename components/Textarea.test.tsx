import { describe, it, expect, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import Textarea, { TextareaProps } from "./Textarea";

const defaultProps: TextareaProps = {
  placeholder: "placeholder",
  value: "value",
  onChange: () => {}, // noop
  readOnly: true,
};

describe("[ components / Textarea ]", () => {

  describe("when Textarea is rendered", () => {
    it("should render", () => {
      const { asFragment } = render(<Textarea {...defaultProps} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("when the value is changed", () => {
    it("should call onChange", () => {
      // Arrange
      const props = {
        ...defaultProps,
        onChange: vi.fn(), // stub
        placeholder: "placeholder to change",
      };

      // Act
      render(<Textarea {...props} />);
      fireEvent.change(screen.getByPlaceholderText("placeholder to change"), {
        target: { value: "new value" },
      });

      // Assert
      expect(props.onChange).toHaveBeenCalled();
    });
  });
});
