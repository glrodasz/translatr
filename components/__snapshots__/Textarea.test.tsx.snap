// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[ components / Textarea ] > when Label is rendered > should render 1`] = `
<DocumentFragment>
  <textarea
    class="w-full mt-4 p-4 bg-gray-100 dark:bg-gray-700 text-gray-900 dark:text-gray-100 rounded resize-none"
    placeholder="placeholder"
    readonly=""
    rows="10"
  >
    value
  </textarea>
</DocumentFragment>
`;

exports[`[ components / Textarea ] > when Textarea is rendered > should render 1`] = `
<DocumentFragment>
  <textarea
    class="w-full mt-4 p-4 bg-gray-100 dark:bg-gray-700 text-gray-900 dark:text-gray-100 rounded resize-none"
    placeholder="placeholder"
    readonly=""
    rows="10"
  >
    value
  </textarea>
</DocumentFragment>
`;
